Apps/Src/SendCarCAN.c

```diff
@@ -1,36 +1,187 @@
 /**
  * @copyright Copyright (c) 2018-2023 UT Longhorn Racing Solar
  * @file SendCarCAN.c
- * @brief Implements the SendCarCAN Task
+ * @brief Function implementations for the SendCarCAN application.
  * 
- * Resends the feedback from the motor controller
+ * This contains functions relevant to placing CAN messages in a CarCAN queue and periodically sending
+ * those messages in the SendCarCAN task.
  * 
  */
 
-#include "Tasks.h"
-#include "CANbus.h"
-#include "CAN_Queue.h"
 #include "common.h"
+#include "os_cfg_app.h"
+#include "CANbus.h"
+#include "Minions.h"
+#include "Contactors.h"
+#include "Pedals.h"
+#include "Tasks.h"
+#include "SendCarCAN.h"
+#include "SendTritium.h"
+
+#define IO_STATE_DLY_MS 250u 
+
+// Task_PutIOState
+OS_TCB putIOState_TCB;
+CPU_STK putIOState_Stk[TASK_SEND_CAR_CAN_STACK_SIZE];
+
+//fifo
+#define FIFO_TYPE CANDATA_t
+#define FIFO_SIZE 256
+#define FIFO_NAME SendCarCAN_Q
+#include "fifo.h"
+
+// Motor message
+#define NUM_MOTOR_MSGS 15 // Messages received from the motor controller
```

Comment:
Might want to leave some comments about how this is used for a sparse array.

Reply:
Yes, thanks for reminding me! Though I was wondering- since the queue isn't actually filling up, do we still want to keep the array? Only half of our messages go out more frequently than 1/sec, and those get sent every 200ms, so I'm not sure how necessary it is to scale them back.

Reply:
If the queue works fine when scaled down (16-32 elements) then might as well only keep the queue. We can always re-add this if there's bandwidth issues on the CAN line.

Reply:
Sounds good